src/worker.js

```diff
@@ -121,16 +130,34 @@ worker.cleanupAndExit = function(code) {
   var _exit = function() {
     worker.exit(code);
   }
-
-  if(!worker.terminationHandler) {
-    return _exit();
+  var _abort = function() {
+    worker.abortListeners = [];
```

Comment:
For a little bit of optimization, can you enclose this in an if statement that checks whether there are listeners and only if so, create a new empty array?